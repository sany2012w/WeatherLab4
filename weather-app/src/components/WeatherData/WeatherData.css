.WeatherBlock {
    margin-top: 30px;
    font-size: 20px;
}

.leftWeather {
  width: 20%;
  float: left;
}

.rightWeather {
  padding-top: 5%;
}

.City {
  font-weight: bold;
  font-size: 20px;
}

.Icon {
  margin-top: 70px;
  margin-left: 10px;
  border: 2px #3d485b ridge;
}

.Temp {
  margin: 10px;
  font-size: 20px;
  font-weight: bold;
}

.Pressure, .Humidity, .Wind, .Clouds, .Coords {
  font-family: sans-serif;
  margin-left: 20%;
  background-color: rgba(197, 204, 213, 0.7);
  background-repeat: no-repeat;
  padding: 1%;
  border-style: ridge;
  border-color: #3d485b;
  border-width: 1px;
}

.button {
  padding: 1%;
  outline: none;
  font-family: 'Bitter', serif;
  color: #21224b;
  border: 2px #3d485b ridge;
  border-radius: 5px;
  padding: 8px;
  font-size: 15px;
  font-weight: bold;
  margin-top: 10px;
}

.button:hover {
  background-color: #586d7c;
  border-color: #c18ce3;
  color: white;
  cursor: pointer;
}

.button:active {
  border-style: inset;
}
