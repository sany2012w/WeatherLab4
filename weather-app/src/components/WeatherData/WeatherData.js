import React from "react";
import './WeatherData.css';
import Loader from '../Loader/Loader';

class WeatherData extends React.Component {
  componentDidMount() {
    if (this.props.onFetch) this.props.onFetch();
  }

  render() {
    if (!this.props.data) {
        return <Loader />
    }

    return (
      <div className="WeatherBlock">
      <div className="City">{this.props.data.name}</div>
        <div className="leftWeather">
          <img className="Icon" src={`https://openweathermap.org/img/wn/${this.props.data.weather[0].icon}.png`} />
          <div className="Temp">{Math.round(this.props.data.main.temp)} Â°C</div>
          {this.props.onDelete && <button className="button" onClick={this.props.onDelete}>Delete</button>}
        </div>

        <div className="rightWeather">
          <div className="Pressure">Pressure: {this.props.data.main.pressure} hPa</div>
          <div className="Humidity">Humidity: {this.props.data.main.humidity}%</div>
          <div className="Clouds">Clouds: {this.props.data.weather[0].description}</div>
          <div className="Wind">Wind: {this.props.data.wind.speed} m/s</div>
          <div className="Coords">Coordinates: [{this.props.data.coord.lon}, {this.props.data.coord.lat}]</div>
        </div>

      </div>
    );
  }
}

export default WeatherData
