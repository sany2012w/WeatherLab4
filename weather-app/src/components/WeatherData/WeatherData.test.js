import React from "react";
import WeatherData from "./WeatherData";
import renderer from "react-test-renderer";
import weatherTestData from "./weatherTestData";

describe("Geolocation weather data", () => {
    it("Weather data loaded", () => {
        const tree = renderer.create(<WeatherData data={weatherTestData}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Weather data is loading", () => {
        const tree = renderer.create(<WeatherData/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

describe("AddCity weather data", () => {
    it("Weather data loaded", () => {
        const tree = renderer.create(<WeatherData
          data={weatherTestData}
          onFetch={()=> {}}
          onDelete={()=> {}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Weather data is loading", () => {
        const tree = renderer.create(<WeatherData
          onFetch={()=> {}}
          onDelete={()=> {}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
