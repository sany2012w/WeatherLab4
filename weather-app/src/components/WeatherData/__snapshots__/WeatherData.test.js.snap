// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCity weather data Weather data is loading 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader-circle"
  />
  <div
    className="loader-bottom"
  >
    Please wait, weather is loading...
  </div>
</div>
`;

exports[`AddCity weather data Weather data loaded 1`] = `
<div
  className="WeatherBlock"
>
  <div
    className="City"
  >
    London
  </div>
  <div
    className="leftWeather"
  >
    <img
      className="Icon"
      src="https://openweathermap.org/img/wn/09d.png"
    />
    <div
      className="Temp"
    >
      7
       °C
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="rightWeather"
  >
    <div
      className="Pressure"
    >
      Pressure: 
      1012
       hPa
    </div>
    <div
      className="Humidity"
    >
      Humidity: 
      81
      %
    </div>
    <div
      className="Clouds"
    >
      Clouds: 
      light intensity drizzle
    </div>
    <div
      className="Wind"
    >
      Wind: 
      4.1
       m/s
    </div>
    <div
      className="Coords"
    >
      Coordinates: [
      -0.13
      , 
      51.51
      ]
    </div>
  </div>
</div>
`;

exports[`Geolocation weather data Weather data is loading 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader-circle"
  />
  <div
    className="loader-bottom"
  >
    Please wait, weather is loading...
  </div>
</div>
`;

exports[`Geolocation weather data Weather data loaded 1`] = `
<div
  className="WeatherBlock"
>
  <div
    className="City"
  >
    London
  </div>
  <div
    className="leftWeather"
  >
    <img
      className="Icon"
      src="https://openweathermap.org/img/wn/09d.png"
    />
    <div
      className="Temp"
    >
      7
       °C
    </div>
  </div>
  <div
    className="rightWeather"
  >
    <div
      className="Pressure"
    >
      Pressure: 
      1012
       hPa
    </div>
    <div
      className="Humidity"
    >
      Humidity: 
      81
      %
    </div>
    <div
      className="Clouds"
    >
      Clouds: 
      light intensity drizzle
    </div>
    <div
      className="Wind"
    >
      Wind: 
      4.1
       m/s
    </div>
    <div
      className="Coords"
    >
      Coordinates: [
      -0.13
      , 
      51.51
      ]
    </div>
  </div>
</div>
`;
